src/components/20-molecules/cookie-disclaimer/index.js

```diff
@@ -0,0 +1,103 @@
+import { LitElement, html, css, unsafeCSS } from 'lit-element';
+
+/* eslint-disable import/no-extraneous-dependencies */
+import defineOnce from '../../../utils/define-once';
+import styles from './index.scss';
+
+const _global = window || global;
+
+class AXACookieDisclaimer extends LitElement {
+  static get tagName() {
+    return 'axa-cookie-disclaimer';
+  }
+
+  static get styles() {
+    return css`
+      ${unsafeCSS(styles)}
+    `;
+  }
+
+  static get properties() {
+    // Define properties and types
+    return {
+      onClick: { type: Function },
+      buttonname: { type: String },
+      title: { type: String },
+      variant: { type: String },
+    };
+  }
+
+  constructor() {
+    super();
+    this.onClick = () => {};
+  }
+
+  firstUpdated() {
+    if (this.hasAccepted()) {
+      this.selfDestruction();
+    } else {
+      const resetStyle = document.createElement('style');
+      resetStyle.textContent = `
+        axa-cookie-disclaimer p {
+          margin: 0;
+          padding: 0;
+        }
+      `;
+      this.appendChild(resetStyle);
+    }
+  }
+
+  handleButtonClick() {
+    const { localStorage } = _global;
+    if (localStorage) {
+      this.selfDestruction();
+      localStorage.setItem(
+        'axa-ch-cookie-disclaimer-accepted',
+        new Date().getTime()
+      );
+    }
+  }
+
+  selfDestruction() {
+    this.parentNode.removeChild(this);
+  }
+
+  hasAccepted() {
+    const { localStorage } = _global;
+    return !!localStorage.getItem('axa-ch-cookie-disclaimer-accepted');
+  }
+
+  render() {
+    const { buttonname, title, variant } = this;
+    const fixed = variant === 'fixed';
```

Comment:
I'ts could be better to use lit html class map directive for performance.
```
const classes = {
   m-cookie-disclaimer: true,
   m-cookie-disclaimer--fixed: variant === 'fixed'
}

class=${classMap(classes)}
```

Reply:
done